File: Back/src/main/java/co/edu/utp/isc/gia/proyectoCrud/Service/PacienteService.java
package co.edu.utp.isc.gia.proyectoCrud.Service;

import co.edu.utp.isc.gia.proyectoCrud.DTO.PacienteDTO;
import co.edu.utp.isc.gia.proyectoCrud.Exceptions.FaltaInfoException;
import co.edu.utp.isc.gia.proyectoCrud.Exceptions.NoExisteException;

import java.util.List;

public interface PacienteService {

    PacienteDTO crearPaciente(PacienteDTO pacienteDTO);
    PacienteDTO consultarPaciente(Long cedulaPaciente) throws NoExisteException;
    PacienteDTO actualizarPaciente(PacienteDTO pacienteDTO) throws NoExisteException;
    PacienteDTO borrarPaciente(Long cedulaPaciente) throws NoExisteException;
    List<PacienteDTO> consultarEstado(String estado) throws NoExisteException;

    List<PacienteDTO> obtenerPacientes();
}

File: Front/src/app/services/paciente.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from 'src/environments/environment';
import { PacienteDTO } from '../model/PacienteDTO';

@Injectable({
  providedIn: 'root'
})
export class PacienteService {

  constructor(private http: HttpClient) { }

  public enviarPaciente(datosPaciente: PacienteDTO): Observable<PacienteDTO> {

    return this.http.post<PacienteDTO>(environment.protocol + environment.localhost + "/paciente/crearPaciente", datosPaciente);
  }

  public obtenerPacientes(): Observable<PacienteDTO[]> {

    return this.http.get<PacienteDTO[]>(environment.protocol + environment.localhost + "/paciente/obtenerPacientes");
  }

  public consultarPaciente(cedulaPaciente: number): Observable<PacienteDTO> {

    return this.http.get<PacienteDTO>(environment.protocol + environment.localhost + "/paciente/consultarPaciente", {"params": {cedulaPaciente: cedulaPaciente}});
  }

  public consultarEstado(estado: string): Observable<PacienteDTO[]> {
    
    return this.http.get<PacienteDTO[]>(environment.protocol + environment.localhost + "/paciente/consultarEstado", {"params": {estado: estado}});
  }

  public actualizarPaciente(datosPaciente: PacienteDTO): Observable<PacienteDTO> {

    return this.http.put<PacienteDTO>(environment.protocol + environment.localhost + "/paciente/actualizarPaciente", datosPaciente);
  }

  public borrarPaciente(cedulaPaciente: number): Observable<PacienteDTO>{

    return this.http.delete<PacienteDTO>(environment.protocol + environment.localhost + "/paciente/borrarPaciente", {"params": {cedulaPaciente: cedulaPaciente}});
  }
}
